examples/ASP.NET Core/README.md

```diff
@@ -0,0 +1,118 @@
+# OpenAI ASP.NET Core Example
+
+This example demonstrates how to use the OpenAI .NET client library with ASP.NET Core's dependency injection container, registering a ChatClient as a singleton for optimal performance and resource usage.
+
+## Features
+
+- **Singleton Registration**: ChatClient registered as singleton in DI container
+- **Thread-Safe**: Demonstrates concurrent usage for chat completion endpoints
+- **Configurable Model**: Model selection via configuration (appsettings.json)
+- **Modern ASP.NET Core**: Uses minimal APIs with async/await patterns
+
+## Prerequisites
+
+- .NET 8.0 or later
+- OpenAI API key
+
+## Setup
+
+1. **Set your OpenAI API key** using one of these methods:
+
+   **Environment Variable (Recommended):**
+
+   ```bash
+   export OPENAI_API_KEY="your-api-key-here"
+   ```
+
+   **Configuration (appsettings.json):**
+
+   ```json
+   {
+     "OpenAI": {
+       "Model": "gpt-4o-mini",
+       "ApiKey": "your-api-key-here"
+     }
+   }
+   ```
+
+2. **Install dependencies:**
+
+   ```bash
+   dotnet restore
+   ```
+
+3. **Run the application:**
+
+   ```bash
+   dotnet run
+   ```
+
+## API Endpoints
+
+### Chat Completion
+
+- **POST** `/chat/complete`
+- **Request Body:**
+
+  ```json
+  {
+    "message": "Hello, how are you?"
+  }
+  ```
+
+- **Response:**
+
+  ```json
+  {
+    "response": "I'm doing well, thank you for asking! How can I help you today?"
+  }
+  ```
+
+## Testing with cURL
+
+**Chat Completion:**
+
+```bash
+curl -X POST "https://localhost:7071/chat/complete" \
```

Comment:
I wonder if we should demonstrate PowerShell or similar as well since cURL isn't baked into Windows.